helm/templates/deploy.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}
+  labels:
+    app.kubernetes.io/name: {{ .Values.name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
```

Comment:
for your labels, consider using a helper function to ensure it's all consistent throughout your chart.
Helm does generate some helper functions when you create a project: https://github.com/bcgov/cas-cif/blob/develop/chart/cas-cif/templates/_helpers.tpl

you can then use them with the [include function](https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-include-function), e.g. `{{ include "cas-cif.labels" . | indent 4 }}`

Reply:
Good idea, will implement. Thanks!

Curious though, whats with the `.` and the `indent 4`? is the `.` there just for context?

Reply:
If my understanding is correct, `.` is the context for the `include` call, yes, and is the [root object](https://docs.helm.sh/docs/chart_template_guide/builtin_objects/) containing built-ins such as `Release` and `Chart`.

The template will generate some text, so you need to indent it 4 spaces to make the YAML valid

Reply:
> The template will generate some text, so you need to indent it 4 spaces to make the YAML valid

of course to make it valid one can't forget those indents